poetry/repositories/legacy_repository.py

```diff
@@ -159,6 +160,15 @@ def __init__(self, name, url, disable_cache=False):
             requests.session(), cache=FileCache(str(self._cache_dir / "_http"))
         )
 
+        url_parts = urlparse.urlparse(self._url)
+        if not url_parts.username:
+            username, password = get_http_basic_auth(self.name)
+            if username and password:
+                netloc = "{}:{}@{}".format(username, password, url_parts.netloc)
```

Comment:
Instead of recreating the url by hand we can just pass the auth information to the session created in `__init__()`. That way the `_url` attribute is left untouched while still providing the auth information to each request made to the remote url.

```python
self._session.auth = get_http_basic_auth(self.name)
```
For this to work we have to make a small change to `get_http_basic_auth()` so that it returns `None` when no authentication information could be found. This makes the code much simpler since we don't have to chech each time if username/password exist. 
```